### ISSUE TITLE
Hard coded CSS indentation (32px) creates Styling glitches on checkout/cart

### ISSUE THREAD

**Issue by gerdneuman** [woocommerce/woocommerce-gateway-stripe#702]
With almost every new release something is broken on the cart and checkout page. Previously left/right floating now this time it is the identation. See here, notice that the Stripe payment method items are indented more than the default WooCommerce methods:

![stripe_checkout](https://user-images.githubusercontent.com/12525563/43353765-176ed998-9240-11e8-87a4-d9870632e5f3.gif)

Basically these two rules make the identation different compared to PayPal  (3rd item) or Direct bank transfer (1st item). Item 2 (Debit) and Item 4 (credit card) are from the stripe plugin.

This rule hard codes 32px identation. Surely this is high risk causing inconsistencies with existing themes which might not all have an exact 32px identation:
```css
.woocommerce-checkout #payment .payment_method_stripe_sepa label[for=payment_method_stripe_sepa] {
    display: block;
    padding-left: 32px;
}
```

This hardcodes `32px`.

Then there's also:
```css
.woocommerce-checkout #payment .payment_method_stripe_sepa, #add_payment_method #payment .payment_method_stripe_sepa {
    position: relative;
}
```

This changes position from default `static` to `relative`.

We use the Virtue theme. Both rules are not in there.

TL;DR: I think it would be better if the plugin would just output the same html markup like is done for the other default payment methods (PayPal, Bank transfer) and leave the CSS styling to the theme. This approach would be much more future proof and compatible.

Just out of memory I think it's the third time I need to update our child theme just because to override these hard coding CSS settings as they are in the Stripe plugin.

:heart:  Otherwise, I would really like to thank you and WooCommerce for a great, clean and reliable plugin!  :heart:

**Comment by lkraav** [comment]
I agree that the styling work here has taken inappropriate risks across versions.

Regardless, it's probably best to just dequeue plugins styles and implement your own.

**Comment by lkraav** [comment]
Right, didn't realize you're the vendor. Anyway, ^^^^ +1

**Comment by webdados** [comment]
This should definitely be removed as it's causing weird behavior with most themes.

`.woocommerce-checkout #payment input#payment_method_stripe, #add_payment_method #payment input#payment_method_stripe {
    position: absolute;
    top: 6px;
}`
and
`.woocommerce-checkout #payment .payment_method_stripe label[for=payment_method_stripe] {
    display: block;
    padding-left: 32px;
}`

No other payment gateway implements this specific CSS, it's just plain nonsense and we have to create !important CSS rules, or dequeue Stripe CSS, on each and every website we implement with Stripe, which makes no sense whatsoever.

**Comment by roykho** [comment]
@gerdneuman @lkraav @webdados - please checkout this branch https://github.com/woocommerce/woocommerce-gateway-stripe/tree/styling

And let me know if that works better for you.

**Comment by gerdneuman** [comment]
@roykho Thanks for working on this!

Unfortunately, not much better:

Before with current version:
![image](https://user-images.githubusercontent.com/12525563/45916229-d5552680-be62-11e8-8012-f75cd145ace1.png)

With the `styling` branch:
![image](https://user-images.githubusercontent.com/12525563/45916230-dede8e80-be62-11e8-8187-ebee7c66eeed.png)

Main two issues are these rules:
```css
.woocommerce-checkout label {
	width: 100%;
}
```
causing the `<label />` to be 100% width and hence too long, which causes the line break between `<input />` and its <label />`. See:

![image](https://user-images.githubusercontent.com/12525563/45916246-3a108100-be63-11e8-89c5-b9e5e2466858.png)


2nd issue is the rule:
```css
.woocommerce-checkout #payment ul.payment_methods li img.stripe-sepa-icon, #add_payment_method #payment ul.payment_methods li img.stripe-sepa-icon {
	float: right;
	max-width: 50px;
}
```
which causes the image to float right. This should just be unset.

We use the Virtue Premium theme.

**Comment by gerdneuman** [comment]
As a side note: Please also test the following pages:

`My Account > Add Payment Method` which currently looks like this:
![image](https://user-images.githubusercontent.com/12525563/45916298-f10cfc80-be63-11e8-89b4-8c8459292f82.png)


and `My account > My subscriptions > Change Payment Method` which looks totally broken with the current version (sorry, cannot test the styling branch on our staging server due to #735 )

![image](https://user-images.githubusercontent.com/12525563/45916395-c15ef400-be65-11e8-853b-c95002e65bdf.png)

As you can see the `input` for the Sepa `label` is at the very top. So it looks really strange. This is due to the rule:
```css
.woocommerce-checkout #payment input#payment_method_stripe, #add_payment_method #payment input#payment_method_stripe {
	position: absolute;
	top: 6px;
}
```
As said this is the current version as I cannot test the branch on our staging server because of #735

**Comment by roykho** [comment]
@gerdneuman by now surely you can appreciate how hard it is to make it work for many different themes.

If we remove all styling, it looks broken in many places. Within minimal styling it breaks in some places. How do we find a happy middle?

We have removed the `position: absolute; top: 6px;` in the styling branch so I am not sure why you're still seeing that.

Also to note we set `label` to 100% because we want the logos to be floated right as it looks weird if it is inline with the text on the left especially when you have many logos like on the credit card.

Can you send me a copy of the theme you're using so I can test against that?

**Comment by gerdneuman** [comment]
> by now surely you can appreciate how hard it is to make it work for many different themes.

Surely I do, sorry if I might have come across any different.

> We have removed the position: absolute; top: 6px; in the styling branch so I am not sure why you're still seeing that.

This was on the current version since I could not test this on staging. But happy to hear this has been resolved.

> Also to note we set label to 100% because we want the logos to be floated right as it looks weird if it is inline with the text on the left especially when you have many logos like on the credit card.

Well, with our theme the logos are on the left by default, that is right next to the label. I personally find this appealing as the eye has to follow only a short distance between label and its icon. See (this is with custom CSS rules to overwrite the stripe plugin css rules):

![image](https://user-images.githubusercontent.com/12525563/45918654-8f13bd80-be8a-11e8-8a69-6925038e43a3.png)

> Can you send me a copy of the theme you're using so I can test against that?

Would love to do. How can I reach you? I did not find an email adress on your github bio or your website.

**Comment by roykho** [comment]
@gerdneuman - This is how it looks right aligned. ![](http://cld.wthms.co/jfF7YG+)

You still feel left align looks better?

roy.ho AT automattic.com

**Comment by gerdneuman** [comment]
Send you the theme. Thanks.

> You still feel left align looks better?

Looks nice. In the end it is a matter of taste probably. Our theme has it left aligned, so for consistency we just stick to this. So if the stripe plugin comes right-aligned and does not leave this to the theme then we have to override this in our child theme. To me it really boils down to what @webdados  said above:

> Besides the credit card form (or any other specific Stripe elements), I don't see why the plugin should implement any styles regarding the INPUT field, the payment method LABEL or the icon IMG tags. The themes already cover that. Any styling on those elements will make Stripe look different from the other payment methods.

Anyway, I do not wanna bikeshed about a float left or right. This one liner would be easy to override in our child theme :)

**Comment by roykho** [comment]
Just received it, thanks @gerdneuman

**Comment by roykho** [comment]
Ok @gerdneuman so I've tweak the styles a little bit more to prevent "over reaching" into theme specific styles. I agree we should let themes handle the styling of this. However it didn't start out this way because there were default themes that didn't have "any" WooCommerce styling as they were not a WooCommerce theme such as TwentySeventeen and we can't have that looking broken.

So with this tweak, we are left only the right alignment. You can test it in this branch https://github.com/woocommerce/woocommerce-gateway-stripe/tree/styling

However I see that the Virtue theme can also benefit from some updates/tweaks like the padding of the list items for each payment is too small or zero thus making it too cramped in my opinion.

Before:
![](http://cld.wthms.co/nntQ18+)

After adding padding:
![](http://cld.wthms.co/KjJgvW+)

Also if you want it to left align like PayPal with added padding:
![](http://cld.wthms.co/KPiIBC+)

Change subscriptions payment:
![](http://cld.wthms.co/DIPcSo+)

**Comment by gerdneuman** [comment]
Looks great, thank you!

**Comment by webdados** [comment]
This plugin should follow the default payment gateways rule and do not add any CSS to payment methods default fields. If it works fine with Storefront, it's OK. Other themes should follow the standards.

**Comment by lkraav** [comment]
> If it works fine with Storefront, it's OK.

Agreed. Twenty Seventeen and friends are not candidates for global compatibility rules. If that's the goal, detect a specific list of themes and enqueue an override stylesheet conditionally. Everything else is mostly an excercise in futility.
